libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1227,39 +1227,64 @@ pair<smtutil::Expression, smtutil::Expression> SMTEncoder::arithmeticOperation(
 	else
 		intType = TypeProvider::uint256();
 
-	smtutil::Expression valueNoMod(
-		_op == Token::Add ? _left + _right :
-		_op == Token::Sub ? _left - _right :
-		_op == Token::Div ? division(_left, _right, *intType) :
-		_op == Token::Mul ? _left * _right :
-		/*_op == Token::Mod*/ _left % _right
-	);
+	optional<smtutil::Expression> valueUnbounded;
+	if (_op == Token::Add)
+		valueUnbounded = _left + _right;
+	else if (_op == Token::Sub)
+		valueUnbounded = _left - _right;
+	else if (_op == Token::Mul)
+		valueUnbounded = _left * _right;
+	else if (_op == Token::Div)
+		valueUnbounded = division(_left, _right, *intType);
+	else /* _op == Token::Mod */
+		valueUnbounded = _left % _right;
 
 	if (_op == Token::Div || _op == Token::Mod)
```

Comment:
We should also keep in mind to investigate whether we should implement those operations themselves with slack variables as well.

Reply:
Yep: https://github.com/ethereum/solidity/issues/9197